extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/MqttThingHandlerFactory.java

```diff
@@ -68,29 +70,37 @@ protected void deactivate(ComponentContext componentContext) {
         super.deactivate(componentContext);
     }
 
+    @Reference
+    protected void setStateDescriptionProvider(MqttChannelStateDescriptionProvider stateDescription) {
+        this.stateDescriptionProvider = stateDescription;
+    }
+
+    protected void unsetStateDescriptionProvider(MqttChannelStateDescriptionProvider stateDescription) {
+        this.stateDescriptionProvider = null;
+    }
+
     @Reference
     protected void setChannelProvider(MqttChannelTypeProvider provider) {
-        this.provider = provider;
+        this.typeProvider = provider;
     }
 
     protected void unsetChannelProvider(MqttChannelTypeProvider provider) {
-        this.provider = null;
+        this.typeProvider = null;
     }
 
     @Override
     protected @Nullable ThingHandler createHandler(@Nullable Thing thing) {
-        MqttChannelTypeProvider provider = this.provider;
-        if (thing == null || provider == null) {
+        if (thing == null) {
```

Comment:
createHandler is defined `protected abstract @Nullable ThingHandler createHandler(Thing thing);` in BaseThingHandlerFactory which is NonNullByDefault so ne need to handle thing == null here.